### PR TITLE
Update/template mvp

### PR DESCRIPTION
**Pull request by waveywaves**
Update includes
* vendored in openshift auth api repo
* Added rest of the resources from the templates

### CODE REVIEW

**Comment by sbose78** [comment]
Does it fix the auth problem?

File: pkg/controller/jenkins/reconciler.go [hunk 1]
```diff
@@ -117,32 +121,49 @@ func (r *JenkinsReconciler) Reconcile(request reconcile.Request) (reconcile.Resu
...
 	jenkinsJNLPSvc := newJenkinsService(instance, jenkinsInstanceName+JenkinsJnlpServiceSuffix, jenkinsJNLPPort) // jenknis jnlp service
-	jenkinsPvc := newJenkinsPvc(instance, jenkinsInstanceName)                                                   // jenknis pvc
+	jenkinsRoute := newJenkinsRoute(instance, jenkinsSvc)
+	jenkinsPvc := newJenkinsPvc(instance, jenkinsInstanceName) // jenknis pvc
+	jenkinsServiceAccount := newJenkinsServiceAccount(instance, jenkinsInstanceName)
+	jenkinsRoleBinding := newJenkinsRoleBinding(instance, jenkinsInstanceName)
 
 	// Set Jenkins instance as the owner and controller
-	if err := controllerutil.SetControllerReference(instance, dc, r.scheme); err != nil {
+	if err := controllerutil.SetControllerReference(instance, jenkinsDc, r.scheme); err != nil {
```
Comment (sbose78) [inline pkg/controller/jenkins/reconciler.go]: Not relevant to this PR, but you should add `ownerReferences` to all objects being created.

File: pkg/controller/jenkins/reconciler.go [hunk 2]
```diff
@@ -117,32 +121,49 @@ func (r *JenkinsReconciler) Reconcile(request reconcile.Request) (reconcile.Resu
...
 	namespacedName := types.NamespacedName{
 		Namespace: found.Namespace,
 		Name:      found.Name,
 	}
 
-	//TODO implements error checking
-	err = r.createResourceIfNotPresent(namespacedName, dc)
+	//TODO implement error checking for existence of the resource
+	// *or if there is an actual error* deal with it (<>_<>-)
+	err = r.createResourceIfNotPresent(namespacedName, jenkinsDc)
```
Comment (sbose78) [inline pkg/controller/jenkins/reconciler.go]: Not related to this PR, but you might want to 
- read the DC if present
- populate relevant parts of it the usual way.
- execute an update/create.

( applies to all objects )

Comment (sbose78) [reply pkg/controller/jenkins/reconciler.go]: Could the user do anything in the CR that might need the owned DC , service account to be updated?

**Comment by DhritiShikhar** [comment]
@waveywaves Can you please share the issue/task for which this PR is raised?

File: pkg/controller/jenkins/reconciler_test.go [hunk 3]
```diff
@@ -28,7 +30,10 @@ func TestNewJenkinsService(t *testing.T) {
 
 		svc := newJenkinsService(mocks.JenkinsCRMock(test_ns, test_name), "test", jenkinsPort)
 		mockSvc := mocks.JenkinsServiceMock(test_ns, test_name)
-
+		b, err := json.MarshalIndent(svc, "", "  ")
+		if err == nil {
+			fmt.Println(string(b))
```
Comment (DhritiShikhar) [inline pkg/controller/jenkins/reconciler_test.go]: Error should be properly logged

File: pkg/controller/jenkins/reconciler.go (hunk 2 above)
Comment (waveywaves) [reply pkg/controller/jenkins/reconciler.go]: Have added a todo to update in the future

File: pkg/controller/jenkins/reconciler.go (hunk 1 above)
Comment (waveywaves) [reply pkg/controller/jenkins/reconciler.go]: done

**Comment by akram** [comment]
/LGTM

that was a good one, well done @waveywaves

**Comment by waveywaves** [comment]
Related to https://github.com/redhat-developer/openshift-jenkins-operator/issues/20

**Comment by akram** [comment]
> Does this PR fix the auth problem?

Yes it does 👍 
The serviceAccount used by the DeploymentConfig is not set at the correct value !

**Pull request merged**